tests/console/yast2_dns_server.pm

```diff
@@ -25,12 +25,24 @@ use version_utils qw(is_leap is_sle sle_version_at_least);
 sub assert_running {
     my $self    = shift;
     my $running = shift;
+    my $cmd     = '(systemctl is-active named || true) | grep -E';
 
     if ($running) {
-        assert_script_run '(systemctl is-active named || true) | grep -E "^active"', timeout => 300;
+        $cmd .= ' "^active"';
+        my $counter = 5;
+        # Service may take a bit of time to actually start
+        while (script_run $cmd) {
```

Comment:
This could run forever. Let's try something different

```perl
my $retries = 5;
my $inactive_on_first;
for (1 .. $retries) {
    my $ret = script_run $cmd;
    last if $ret == 0;
    diag "Service is not active, retry $_ of $retries";
    $inactive_on_first = 1;
    sleep 5;
}
die 'named service is not active even after 25s delay' if $ret != 0;
record_soft_failure 'bsc#1093029' if $inactive_on_first;
```

Alternatively

```perl
my $counter = 5;
while (script_run $cmd) {
    sleep 5;
    $counter--;
    die '…' if $counter == 0;
}
```

Reply:
Ouch. Result of last minute change =)